GVFS/GVFS.Platform.Mac/MacFileBasedLock.cs

```diff
@@ -1,133 +1,17 @@
-                {
-                    // Failures of close() are logged for diagnostic purposes only.
-                    // It's possible that errors from a previous operation (e.g. write(2))
-                    // are only reported in close().  We should *not* retry the close() if
-                    // it fails since it may cause a re-used file descriptor from another
-                    // thread to be closed.
-
-                    int errno = Marshal.GetLastWin32Error();
-                    EventMetadata metadata = this.CreateEventMetadata(errno);
-                    this.Tracer.RelatedWarning(
-                        metadata,
-                        $"{nameof(MacFileBasedLock)}.{nameof(this.Dispose)}: Error when closing lock fd");
-                }
-
-                this.lockFileDescriptor = NativeMethods.InvalidFileDescriptor;
-            }
-        }
-
-        private EventMetadata CreateEventMetadata(int errno = 0)
-        {
-            EventMetadata metadata = new EventMetadata();
-            metadata.Add("Area", "MacFileBasedLock");
-            metadata.Add(nameof(this.LockPath), this.LockPath);
-            if (errno != 0)
-            {
-                metadata.Add(nameof(errno), errno);
-            }
-
-            return metadata;
-        }
-
-        private static class NativeMethods
-        {
-            // #define O_WRONLY    0x0001      /* open for writing only */
-            public const int OpenWriteOnly = 0x0001;
-
-            // #define O_CREAT     0x0200      /* create if nonexistant */
-            public const int OpenCreate = 0x0200;
-
-            // #define EINTR       4       /* Interrupted system call */
-            public const int EIntr = 4;
-
-            // #define EAGAIN      35      /* Resource temporarily unavailable */
-            // #define EWOULDBLOCK EAGAIN  /* Operation would block */
-            public const int EWouldBlock = 35;
-
-            public const int LockSh = 1; // #define LOCK_SH   1    /* shared lock */
-            public const int LockEx = 2; // #define LOCK_EX   2    /* exclusive lock */
-            public const int LockNb = 4; // #define LOCK_NB   4    /* don't block when locking */
-            public const int LockUn = 8; // #define LOCK_UN   8    /* unlock */
-
-            public const int InvalidFileDescriptor = -1;
-
-            public static readonly ushort FileMode644 = Convert.ToUInt16("644", 8);
-
-            [DllImport("libc", EntryPoint = "open", SetLastError = true)]
-            public static extern int Open(string pathname, int flags, ushort mode);
-
-            [DllImport("libc", EntryPoint = "close", SetLastError = true)]
-            public static extern int Close(int fd);
-
-            [DllImport("libc", EntryPoint = "flock", SetLastError = true)]
-            public static extern int FLock(int fd, int operation);
-        }
-    }
-}
+namespace GVFS.Platform.Mac
+{
+    public class MacFileBasedLock : POSIXFileBasedLock
```

Comment:
Seems like, at least right now, this is just a `POSIXFileBasedLock`. Do you know of something coming shortly that will require a distinction?

Reply:
Indeed; see https://github.com/Microsoft/VFSForGit/pull/926#discussion_r271140642. There's no strict reason to abstract it at all just yet, so I'll undo it. That can come later.

Reply:
I'm fine leaving `MacFileBasedLock` as you have it here if we already know why we need it.

Reply:
👍  I'd prefer to start more correct, which I think this way is. There's unlikely to be churn here, too, so it's not an issue.